### ISSUE TITLE
update when?

### ISSUE THREAD

**Issue by UpgradeDenied** [kgrossnickle/eth-box#3]
title

**Comment by VioSynthax** [comment]
It's pointless to abuse the issues system to beg for an update. There won't be any real improvement until someone who intimately understands DirectX shaders decides to spend an unreasonable amount of effort writing one for meager crypto gains. Not going to happen. Buy a miner if you want to mine.

**Comment by UpgradeDenied** [comment]
the xbox series x is pretty comparable to an rtx 2080ti so it _could_ be viable lol + its just pretty cool

**Comment by UpgradeDenied** [comment]
sorry for using the issue system but it was the only way you can _really_ contact the dev

**Comment by VioSynthax** [comment]
The architecture is so different that it's hard to tell what its mining potential could be, it should be closest to an AMD GPU with the same speed of VRAM, so one of the RDNA 2 cards should be close. Theoretically, slightly above a 2080 Ti. Maybe gross $2.50 a day. Free vending machine snacks I suppose.

**Comment by UpgradeDenied** [comment]
$2.5 x 360 = $900

**Comment by UpgradeDenied** [comment]
it would be a shame for this project to die tho, looks cool

**Comment by UpgradeDenied** [comment]
there's web miners but they really don't do much

**Comment by UpgradeDenied** [comment]
getting that $2.5 a day would probole need major refactoring of the code as said in the readme.md

**Comment by VioSynthax** [comment]
If you can get them at MSRP they would have an actually reasonable ROI, but I doubt these will ever reach their full potential until someone inevitably slaps Linux on one. Probably well after they're profitable. Would likely destroy the supply chain even harder if anyone managed to eek out a 7 month ROI from one.

**Comment by UpgradeDenied** [comment]
lol yh, ethhash-hlsl is quite old and hasn't been updated recently, which is what this project is based on. if you have any experience you could try make a fork

**Comment by UpgradeDenied** [comment]
on the subject of forks, ill check them now

**Comment by UpgradeDenied** [comment]
no luck, all the forks are up to date

**Comment by UpgradeDenied** [comment]
linux would require custom kernal code and accounting all the xbox-specific things, it's been done on ps4 sooooo maybe one day

**Comment by VioSynthax** [comment]
PS5 is probably a better mining platform, Linux should be out within the year, assuming the kernel exploit for it gains an entry point.

**Comment by UpgradeDenied** [comment]
![copium](https://user-images.githubusercontent.com/55844307/151835484-2d2696b7-7316-4384-99c7-69fdb9f9f020.jpg)

**Comment by UpgradeDenied** [comment]
pls dev, update

**Comment by UpgradeDenied** [comment]
why is the xbox one / xbox series so unhackable

**Comment by VioSynthax** [comment]
> why is the xbox one / xbox series so unhackable

It's not, just no one with the skills and/or knowledge wants to do it. There have been plenty of bugs that could be used for it. Bugs for Hyper-V, Windows, Edge, Chromium, amd64 architecture, and the AMD PSP could all be potentially used as part of an exploit chain. Microsoft's custom security chip they've used since the One is allegedly coming to PCs "soon" so maybe it will have some attention given to it as well. Theoretically, there have already been enough exploits released to have a fairly thorough unlocking of dev mode, probably even escalation. There's just no scene for it.

**Comment by UpgradeDenied** [comment]
yh i guess, do you have an xbox on a side note?

**Comment by VioSynthax** [comment]
A handful of them, yes.

**Comment by UpgradeDenied** [comment]
i gtg for a  bit but ill b back soon

**Comment by UpgradeDenied** [comment]
mabey scanning throught OSU1 would help? https://support.xbox.com/en-GB/help/hardware-network/console/system-update-solution/offline-system-update

**Comment by UpgradeDenied** [comment]
dev mode is not enough for linux, we need to go deeper

**Comment by VioSynthax** [comment]
> mabey scanning throught OSU1 would help? https://support.xbox.com/en-GB/help/hardware-network/console/system-update-solution/offline-system-update

No, nothing publicly released is going to be useful, anything that could be is encrypted. Only way to get the decrypted versions would be to hack the console. Gotta do enough of it blind to get a dump.

**Comment by UpgradeDenied** [comment]
a dump would be amazing, just how tho...

**Comment by VioSynthax** [comment]
> dev mode is not enough for linux, we need to go deeper

Without some very deep knowledge and some special equipment, that's not happening. Hardware hacking is likely the only feasible method like with the 360 and the Switch, and Microsoft has hardened it against physical attacks /greatly/

**Comment by UpgradeDenied** [comment]
mabey the etehrnet port has a vunrability

**Comment by UpgradeDenied** [comment]
btw feel free to add me on discord since this probally isint the best play to talk discoflea#3083

**Comment by VioSynthax** [comment]
> mabey the etehrnet port has a vunrability

Extraordinarily unlikely, DMA could be promising though. The microsoldering involved scares off most people from that though. These things aren't getting blown open anytime soon.

**Comment by UpgradeDenied** [comment]
mabey the expanision slot?

**Comment by kgrossnickle** [comment]
Most of what you said is correct but two things I'll point out: 

1. Its not "meager crypto gains". TBH I thought someone would already release a miner (either open source through this or close source and use this as baseline) by now because simple math is $50 minimum a month per xbox, someone takes 10$ off the top, get 10k people to use the miner and the dev gets 50,000$ a month :D

2. Doesnt take thattttt much shader knowledge. Any good dev can do it if they have an AMD GPU and some patience. You really only need basics of shader knowledge since the codes there, and just need to trace/ debug / improve.


I don't have much bandwidth to work on this at the moment. The only way for me to improve this is to buy an AMD card ( since that what the xbox has) and debug with the AMD card, since I have an NVIDIA. However spending $500 on a minable AMD GPU just for a possibility of  working isnt worth it for me. My NVIDIA card is ~ 75% max hashrate (28 MH/s vs 35 on regular miner) so I think it's possible to get at least ~50% max hashrate on AMD IMO, maybe 100% but unlikely

**Comment by UpgradeDenied** [comment]
it would be nice to have some bug fixes here and there and also a compiled version... you could debug it on a server (i'm not sure if AWS has GPU servers but that could be an option). My original post was just to check if the project was dead or not lol

**Comment by UpgradeDenied** [comment]
oh and does nicehash work?
